lib/pages/my_reports_pages/components/reports_list_widget.dart

```diff
@@ -1,98 +1,545 @@
 import 'package:flutter/material.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:intl/intl.dart';
+import 'package:mosquito_alert_app/api/api.dart';
+import 'package:mosquito_alert_app/models/owcampaing.dart';
 import 'package:mosquito_alert_app/models/report.dart';
 import 'package:mosquito_alert_app/utils/MyLocalizations.dart';
 import 'package:mosquito_alert_app/utils/Utils.dart';
+import 'package:mosquito_alert_app/utils/customModalBottomSheet.dart';
 import 'package:mosquito_alert_app/utils/style.dart';
-import 'package:intl/intl.dart';
 
-class ReportsList extends StatelessWidget {
-  final Function onTap;
+
+class ReportsList extends StatefulWidget {
+  ReportsList({Key? key, required this.reports}) : super(key: key);
   final List<Report> reports;
 
-  ReportsList(this.reports, this.onTap);
+  @override
+  _MyReportsListState createState() => _MyReportsListState(reports: reports);
+}
+
+class _MyReportsListState extends State<ReportsList> {
+  List<Report> reports;
+  GoogleMapController? miniMapController;
+
+  _MyReportsListState({required this.reports});
+
+  String formatCreationTime(String? utcTimeString) {
+    utcTimeString ??= '1970-01-01 00:00';
+    var utcTime = DateTime.parse(utcTimeString);
+    var localTime = utcTime.toLocal();
+    var formattedTime = DateFormat('yyyy-MM-dd HH:mm').format(localTime);
+    
+    return formattedTime;
+  }
+
+  String getTitle(BuildContext context, Report report){
+    switch(report.type) {
+      case 'adult':
+        var response_question_6 = report.responses?.firstWhere(
+          (element) => element!.question == 'question_6');
+
+        if (response_question_6 == null){
+          return '';
+        }
+
+        if (response_question_6.answer == 'question_6_answer_64'){
+          return 'Non identified';
+        }
+
+        return MyLocalizations.of(context, response_question_6.answer);
+      case 'bite':
+        var response_question_1 = report.responses?.firstWhere(
+          (element) => element!.question == 'question_1');
+
+        var numBites = response_question_1?.answer_value;
+        if (numBites == null){
+          return '';
+        }
+        if (numBites == '1'){
+          return '$numBites bite';
```

Comment:
You can use the translation of the string 'single_bite' for these, will be wrong in determining singular/plural forms thought.

`return "$numBites MyLocalizations.of(context, 'single_bite')"`

Reply:
will we have a translation string also for the plural? so that is not hardcoded for plural and for 0

Reply:
I'm creating a new string in localize for it: 'plural_bite'

Reply:
Just make sure to lowercase the result in both cases:
OK: 3 bites
OK: 1 bite
KO: 3 Bites

Reply:
Already created in localise.biz

Reply:
I've added it now, in lowercase and also downloaded the strings from localise.biz